packages/babel-types/src/ast-types/generated/index.ts

```diff
@@ -1905,7 +1905,7 @@ export interface TSMappedType extends BaseNode {
 
 export interface TSLiteralType extends BaseNode {
   type: "TSLiteralType";
-  literal: NumericLiteral | StringLiteral | BooleanLiteral | BigIntLiteral;
+  literal: any;
```

Comment:
The automatically generated type does not look right, should I manually correct it?

Reply:
This file is generated from `packages/babel-types/scripts/generators/ast-types.js`, we can add a manual allowlist of handwritten types for functional validators.

Reply:
I added a `oneOf` allowlist for validator, now it seems good!